main.js

```diff
@@ -79,26 +94,64 @@ define(function (require, exports, module) {
         }
         
         // Get size of the SVG image (which is always explicitly specified on root tag)
-        var svgWidth, svgHeight;
-        var viewBoxAttr = $svgRoot[0].getAttribute("viewBox"); // jQ can't read this attr - see below
-        if (viewBoxAttr) {
-            var boundsStrs = viewBoxAttr.split(/[ ,]+/);
-            svgWidth = parseFloat(boundsStrs[2]);
-            svgHeight = parseFloat(boundsStrs[3]);
-        } else {
-            svgWidth = parseInt($svgRoot.attr("width"), 10);
-            svgHeight = parseInt($svgRoot.attr("height"), 10);
+        // First option: Use explicit height and width (with units!) from file
+        // if either is not available, or if they use percentages, then
+        // Second option: Use viewBox width and height (in pixels) 
+        //                scaling by aspect ratio if one of width *or* height specified
+        // Third option: Use HTML5 default size (300px x 150px)
+        var svgWidth, svgHeight, unit;
+        var svgWidthAttr  = $svgRoot.attr("width");
+        var svgHeightAttr = $svgRoot.attr("height");
+        var viewBoxAttr   = $svgRoot[0].getAttribute("viewBox"); 
+                            // jQ can't read this attr - see below
+        if (svgWidthAttr) {
+            svgWidth = parseFloat(svgWidthAttr);
+            unit = svgWidthAttr.match(/[a-z%]+/);
+            if (unit) {
+                //Multiply by unit value in pixels.
+                //Will result in NaN if unit not supported,
+                //causing a fall-through to another sizing option
+                svgWidth *= unitsToPixels[ unit[0] ];
+            }
+        }
+        if (svgHeightAttr) {
+            svgHeight = parseFloat(svgHeightAttr);
+            unit = svgHeightAttr.match(/[a-z%]+/);
+            if (unit) {
+                svgHeight *= unitsToPixels[ unit[0] ];
+            }            
         }
+        if (viewBoxAttr) {
+            var bounds = viewBoxAttr.split(/[ ,]+/).map(parseFloat);
+            if (!svgWidth) {
+                svgWidth = svgHeight ? svgHeight * bounds[2] / bounds[3]
+                            : bounds[2];
+            }
+            if (!svgHeight) {
+                svgHeight = svgWidth ? svgWidth * bounds[3] / bounds[2]
+                            : bounds[3];
+            }
+        } 
+        // If either has not been defined, or is zero, set defaults:
+        svgWidth  = svgWidth  || 300;   
+        svgHeight = svgHeight || 150;
+        
         
         // Specify actual width consistent with zoom factor
         var viewWidth = svgWidth * currentState.zoomFactor;
         var viewHeight = svgHeight * currentState.zoomFactor;
         
         // jQ auto lowercases the attr name, making it ignored (http://bugs.jquery.com/ticket/11166 resolved wontfix: "we don't support SVG")
-        $svgRoot[0].setAttribute("viewBox", "0 0 " + svgWidth + " " + svgHeight);
+        if (!viewBoxAttr) {
```

Comment:
Sorry, never mind -- this is correct as-is.

I was tripped by the case where viewBox _and_ width/height are already specified in the original SVG, applying an initial scale/zoom factor out of the box.  It's a bit ambiguous what the zoom feature here should call "100%": the SVG size given by the width/height, or the "natural" size if only the viewBox attribute were present without any scaling due to the additional width/height set.  Currently, it calls the specified width/height (with scaling included) 100%, which seems fair to me.